### ISSUE TITLE
[DOC] Segment replication changes with remote store GA and 2.10 release

### ISSUE THREAD

**Issue by dreamer-89** [opensearch-project/documentation-website#5011]
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** 
With 2.10.0, [remote store feature](https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/remote-store/index/#segment-replication-and-remote-backed-storage) is going GA which by default works only with segment replication feature. With this, segment replication now available in two flavours
1.  With remote store - primary uploads to remote store, replica downloads from remote store.  
2.  Without remote store where node perform node-node communication for segment downloads (existing). 

Existing segment replication documentation is applicable only for 2 above and thus needs amendments to incorporate remote store availability. Below section shows the changes and new documentation that can be added/udpated. 

#### Considerations
Update segment replication [considerations](https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/segment-replication/index/#considerations) documentation
1. Integration with remote-backed storage as the source of replication is currently not supported. --> Delete this as remote backed storage as replication source is now supported
5. Segment replication leads to increased network congestion on primary shards. --> This is only applicable for node-node communication. With remote store, primary shard uploads on remote store while replica copies downloads directly from remote store. 
6. Read-after-write guarantees: The wait_until refresh policy is not compatible with segment replication. If you use the wait_until refresh policy while ingesting documents, you’ll get a response only after the primary node has refreshed and made those documents searchable. Replica shards will respond only after having written to their local translog. We are exploring other mechanisms for providing read-after-write guarantees. For more information, see the corresponding GitHub issue. --> Read after write (RAW) guarantees are now supported using get/mget or using _primary shard preference based search. Related github issue https://github.com/opensearch-project/OpenSearch/issues/8700, https://github.com/opensearch-project/OpenSearch/issues/6046
7. System indexes will continue to use document replication internally until read-after-write guarantees are available. In this case, document replication does not hinder the overall performance because there are few system indexes. --> System indices are now supported with segment replication. 

#### Configurations
Update segment replication [configuration](https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/segment-replication/index/#considerations) documentation
1. In segment replication, the primary shard is usually generating more network traffic than the replicas because it copies segment files to the replicas. Thus, it’s beneficial to distribute primary shards equally between the nodes. To ensure balanced primary shard distribution, set the dynamic cluster.routing.allocation.balance.prefer_primary setting to true. --> This is applicable only for node-node communication.
2. This setting is not applied to system indexes and hidden indexes. By default, all system and hidden indexes in OpenSearch will still use document replication even if this setting is enabled. --> Delete this `note`, as this is not true anymore. 

### Add new documentation
Add references to [remote store feature](https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/remote-store/index/#segment-replication-and-remote-backed-storage) in segment replication documentation as alternate mechanism for segment syncs. Along with existing example on index creation, we can
1. Add another example which shows remote store related settings used during index creation. 
2. (Better) Refer to existing remote store documentation which already contains [create index](https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/remote-store/index/#create-an-index) example

```
PUT /my-index1
{
  "settings": {
    "index": {
      "replication.type": "SEGMENT" 
    }
  }
}
```

OS Version. 2.10.0+

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

**Comment by Naarcha-AWS** [comment]
@dreamer-89: Is this in addition to or a duplicate of #4875, #4984, and #4405?

**Comment by dreamer-89** [comment]
> @dreamer-89: Is this in addition to or a duplicate of #4875, #4984, and #4405?

Thanks @Naarcha-AWS for the update. Yes, #4405 seems to be duplicate of this issue. We can use details/changes mentioned in this issue and close this issue in favour of #4405. #4875 - seems specific to remote store GA release while #4984 is about adding new consideration for get/mget, term-vector APIs.
